compiler/rustc_parse/src/parser/expr.rs

```diff
@@ -1759,14 +1775,52 @@ impl<'a> Parser<'a> {
             }
         }
 
-        let token = recovered.as_ref().unwrap_or(&self.token);
-        match Lit::from_token(token) {
+        // How many leading invisible delimiters are there of the right kind?
```

Comment:
If we want to parse a literal and see an invisible delimiter, then we probably just parse an expression and then report an error if the resulting expression is not a literal.

Not sure whether it should be done here in `fn parse_opt_lit` or 1-2 levels above.